include/envoy/local_info/local_info.h

```diff
@@ -25,17 +25,17 @@ class LocalInfo {
   /**
    * Human readable zone name. E.g., "us-east-1a".
    */
-  virtual const std::string& zoneName() const PURE;
+  virtual const std::string zoneName() const PURE;
```

Comment:
Yeah, I agree that this is the best we can do for now. This will be eventually resolved as the types converge.